src/include/processor/operator/recursive_extend/path_scanner.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "bfs_state.h"
+
+namespace kuzu {
+namespace processor {
+
+// FixedSizePathScanner scans all path for a fixed length.
+struct FixedSizePathScanner {
+    const BaseBFSMorsel& bfsMorsel;
+    // Number of extension performed during recursive join
+    size_t depth;
+    // Current path. TODO(Xiyang): we can remove this and directly read from cursor and nbrs.
+    std::vector<common::offset_t> path;
+
+    // States
+    std::vector<size_t> cursorPerDepth;
+    std::vector<std::vector<common::offset_t>*> nbrsPerDepth;
+    common::offset_t currentDstOffset;
+
+    FixedSizePathScanner(const BaseBFSMorsel& bfsMorsel, size_t depth)
```

Comment:
It's confusing that FixedSizePathScanner takes a BFSMorsel. Can you change this so it's more clear what's really neeed?

Reply:
Leave a comment but decide not to change this yet.